### PR TITLE
V2.0 Shell - Progressive Rendering with Skeleton

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes # (issue)
[Demo Loom](https://www.loom.com/share/143c8669b07f4d9fb5e835c6537e0919)
Also Fixes
- Routing Forms Navigation is not shown as current for Form Builder and Forms list pages
- Fixes move up and down buttons as per new UI
Screenshot 2022-09-03 at 5 15 11 PM

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
this doesnt look good, just remove that skeleton here
CleanShot 2022-09-05 at 10 41 49@2x

**Comment by hariombalhara** [comment]
@PeerRich Done. Though this will introduce CLS.

**Comment by PeerRich** [comment]
> @PeerRich Done. Though this will introduce CLS.

I see. maybe instead we wanna have the skeleton as the same size as the content? have one skeleton left for headline and one small one for the button on the right

**Comment by hariombalhara** [comment]
Yeah, that's some extra work just for the skeleton. By default, if no content is there in the element, it just collapses. We need a way to not have the element collapse. 
Just thought maybe we can have opacity 0 text in the skeleton element, that would not let it collapse and it won't be visible. This would allow us to just keep using SkeletonText without explicitly hardcoding min-height

**Comment by PeerRich** [comment]
> Yeah, that's some extra work just for the skeleton. By default, if no content is there in the element, it just collapses. We need a way to not have the element collapse. Just thought maybe we can have opacity 0 text in the skeleton element, that would not let it collapse and it won't be visible. This would allow us to just keep using SkeletonText without explicitly hardcoding min-height

that's a good approach

File: apps/web/components/eventtype/CreateEventType.tsx [hunk 1]
```diff
@@ -51,7 +52,7 @@ interface CreateEventTypeBtnProps {
 export default function CreateEventTypeButton(props: CreateEventTypeBtnProps) {
   const { t } = useLocale();
```
Comment (zomars) [inline apps/web/components/eventtype/CreateEventType.tsx:53]: @hariombalhara why not use this instead?

```suggestion
  const { t, isLocaleReady } = useLocale();
```

File: apps/web/components/I18nLanguageHandler.tsx [hunk 2]
```diff
@@ -14,6 +14,11 @@ export function useViewerI18n() {
   });
 }
 
+export function useIsI18nLoading() {
+  const i18n = useViewerI18n();
+  return i18n.status === "loading";
+}
+
```
Comment (zomars) [inline apps/web/components/I18nLanguageHandler.tsx:21]: [See my comment](https://github.com/calcom/cal.com/pull/4138/files#r963085478)

**Review by zomars — APPROVED** [review]
Removed  `isI18nLoading` in favor of  `isLocaleReady`, tested and it seems good to go. Thanks @hariombalhara please DM me if you feel this was a bad move.

**Pull request merged**